pkg/minikube/config/node.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+// AddNode adds a new node config to an existing cluster.
+func AddNode(cc *MachineConfig, name string, controlPlane bool, k8sVersion string, profileName string) error {
+	t := Worker
```

Comment:
Without context, this feels somewhat confusing (What is worker? What is T going to be used for?). What do you feel about introducing the context first by supplying a default value?

```
n := Node{Name: name, Type: Worker}
if controlPlane {
  n.Type = Master
}
```

Reply:
Master is a fuzzy term in Kubernetes, and isn't described in the official glossary any longer:  https://kubernetes.io/docs/reference/glossary/?fundamental=true

Since machines can be both workers and run the control plane, should the node object be able to describe all 3 possibilities?

For example:

```
Node.ControlPlane bool 
Node.Worker bool
```

That may be easier than working with an array of roles. It does however mean you could end up with a Node that isn't a worker, which would be a bit odd. Do whatever you feel is most readable.